src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,119 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+
+/* eslint-disable jsx-a11y/control-has-associated-label */
+type Props = {
+  todo: Todo,
+  toggleTodoStatus: (todoId: number) => void,
+  deleteTodo: (todoId: number) => void,
+  editTodo: (editedTitle: string, todoId: number) => void,
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  toggleTodoStatus,
+  deleteTodo,
+  editTodo,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [editText, setEditText] = useState(todo.title);
+
+  const submitHandler = (
+    event: React.FocusEvent | React.KeyboardEvent,
+  ) => {
+    if ('key' in event) {
+      if (event.key === 'Enter' || event.key === 'Escape') {
+        if (!editText) {
+          deleteTodo(todo.id);
+        }
+      }
+
+      switch (event.key) {
+        case 'Enter':
+          editTodo(editText, todo.id);
+          setIsEditing(false);
+          break;
+
+        case 'Escape':
+          setEditText(todo.title);
+          setIsEditing(false);
+          break;
+
+        default:
+          break;
+      }
+    } else {
+      if (!editText) {
+        deleteTodo(todo.id);
+      }
```

Comment:
nested conditions – no, no, and no

Reply:
also, as you can see, `setIsEditing(false)` is performed every time, so you should take it out of these conditions